src/cart/crt_hg.c

```diff
@@ -1093,27 +1095,15 @@ crt_hg_req_send(struct crt_rpc_priv *rpc_priv)
 			  rpc_priv->crp_tgt_uri);
 	}
 
-	/* HG_Forward can return 2 types of errors - network errors and generic
-	 * ones, such as out of memory, bad parameters passed, invalid handle
-	 * etc...
-	 *
-	 * For network errors, we do not want to return error back to the caller
-	 * of crt_req_send(), but instead we want to invoke completion callback
-	 * manually with 'node unreachable' (DER_UNREACH) error.
-	 *
-	 * HG_NA_ERROR and HG_PROTOCOL_ERROR are both network-level errors
-	 * that can be raised by mercury.
-	 */
-	if (crt_hg_network_error(hg_ret)) {
+	/* For any error to be reported via completion callback */
+	if (hg_ret != HG_SUCCESS) {
```

Comment:
do we want to differentiate errors, e.g upper level stack might want to retry for HG_TIMEOUT or HG_AGAIN, but it does not make sense to retry for HG_PERMISSION or HG_PROTOCOL_ERROR

Reply:
I ll make a note to add propagation of error into callback to let caller differentiate between errors